library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```diff
@@ -84,18 +84,15 @@ long getBatchSizeInBytes(long batchId) {
         long totalSize = 0;
         try {
             String[] selectionArgs = {String.valueOf(batchId)};
-            cursor = resolver.query(ALL_DOWNLOADS_CONTENT_URI,
-                    null,
+            cursor = resolver.query(
+                    ALL_DOWNLOADS_CONTENT_URI,
+                    new String[]{"sum(" + COLUMN_TOTAL_BYTES + ")"},
                     COLUMN_BATCH_ID + " = ?",
                     selectionArgs,
                     null);
 
-            int totalBytesColumnIndex = cursor.getColumnIndexOrThrow(COLUMN_TOTAL_BYTES);
-
-            while (cursor.moveToNext()) {
-                int individualDownloadSize = cursor.getInt(totalBytesColumnIndex);
-                totalSize += individualDownloadSize;
-            }
+            cursor.moveToFirst();
+            totalSize = cursor.getLong(0);
```

Comment:
It is the column index of the projection.


  [image: --]
Friedger Müffke
[image: http://]about.me/friedger
<http://about.me/friedger>


2015-07-01 12:00 GMT+02:00 Adam Brown <notifications@github.com>:

> In
> library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java
> <https://github.com/novoda/download-manager/pull/49#discussion_r33664565>:
>
> >                      COLUMN_BATCH_ID + " = ?",
> >                      selectionArgs,
> >                      null);
> >
> > -            int totalBytesColumnIndex = cursor.getColumnIndexOrThrow(COLUMN_TOTAL_BYTES);
> > -
> > -            while (cursor.moveToNext()) {
> > -                int individualDownloadSize = cursor.getInt(totalBytesColumnIndex);
> > -                totalSize += individualDownloadSize;
> > -            }
> > +            cursor.moveToFirst();
> > +            totalSize = cursor.getLong(0);
>
> what is this column id? I thought you needed to define another column
> variable in the raw query and then read from that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/download-manager/pull/49/files#r33664565>.
>